### DISCUSSION TITLE
Archcraft OS

### DISCUSSION THREAD

**Discussion by vanillyn** [catppuccin/catppuccin#1935] [Port Requests]
### Will you make the port?

- [X] Yes, I'm willing to make the port

### How can the requested port be themed?

Archcraft theming is done in Openbox. The creator would configure the files in ~/.config/openbox-themes with the custom colors for the bar and menu, and then would configure the theme files in /usr/share/themes. Openbox config is in /usr/share/themes/openbox-3, GTK config is in /usr/share/themes/gtk-n, and other themes are on ~/.config/openbox-themes/theme/program.
Archcraft is configured to automatically update the themes from the configuration files to make theming simpler.

### Has the requested port been themed already?

I've created the port here
- https://github.com/vanillyn/archcraft

![archcraft](https://user-images.githubusercontent.com/108785622/220466915-a29c7135-c46a-4799-bf90-bf8398f27e8c.png)

### Any additional comments?

Archcraft OS is a special OS that's meant to be pretty and stylistic, but not everyone can use it, so I've ported the Archcraft files to Arch Linux.

This port uses other Catppuccin ports, so it is not entirely unique.
Themes used: [GTK](https://github.com/catppuccin/gtk)
The other configuration is unique.

Only an Openbox version is available now, although Archcraft uses several WM's. Porting to bspwm will be done soon.

**Comment by nekowinston** [comment]
I worry about maintainability with this port. https://github.com/catppuccin/regolith was only moved over to the org, once it added both https://github.com/catppuccin/gtk and https://github.com/catppuccin/gnome-terminal as either submodules or in the install script.
Regolith also has a simple color definition file (e.g. [`colors.mocha`](https://github.com/catppuccin/regolith/blob/main/catppuccin/flavours/colors.mocha)), that generates the other i3 & menu themes.

Here, `#1e1e2e` appears in 14 files, most of them are GTK, but it also appears 33 times in [`themerc`](https://github.com/vanillyn/archcraft/blob/93b13de1191ec05b7fae25c2ea8936bbd2601b03/openbox/Mocha/root/usr/share/themes/Catppuccin-Mocha/openbox-3/themerc).

**Reply by vanillyn** [reply]
Yeah, a lot of my configuration was done in Arch Linux, a lot of stuff was different. I'm of course still working on it, though. I've been trying to share updates.
Archcraft has all of the color definitions in the theme.bash file, and I'm of course going to be using it more often.
Also, yeah, I should probably use GTK as a submodule, but I also had to configure it in order to match Archcraft's design, in order to not change it too drastically. But I could also try that.

**Reply by nekowinston** [reply]
You could apply patches over the GTK subrepo like this: https://github.com/nekowinston/ctp-gtk-patch/blob/main/build.sh
This particular way of applying the same patch over all four flavors is pretty horrible, it's just to show the general way of doing it.

**Reply by vanillyn** [reply]
its weird, python script just isn't installing anything
im trying the script, and am editing it to match the needs

**Comment by vanillyn** [comment]
ive finished optimizing the theme btw
im now working on adding separate colors

**Comment by vanillyn** [comment]
ill take a break from the project or now, if you decide to drop it, let me know, but ill take a bit of time off

**Comment by vanillyn** [comment]
i've finished fixing the script!
apparently the gtk theme wasn't installing correctly
ive also added the openbox-3 folder to the themes, as it wasn't automatically added with the gtk theme, causing window decorations to not appear
the openbox-3 themerc files can not be themed with the theme.bash file, so ive had to apply the hex colors there manually

all in all, everything should be working for openbox, and all of archcraft has been themed

**Reply by vanillyn** [reply]
i should work on making sure the color balance is right and matches the screenshots

**Comment by sgoudham** [comment]
Closed as #2019 has been raised
